fenix/app/src/main/java/org/mozilla/fenix/shopping/ui/ReviewQualityCheckMessageCard.kt

```diff
@@ -0,0 +1,215 @@
+ * @param modifier Modifier to be applied to the card.
+ * @param description The optional secondary piece of text.
+ * @param linkText An optional piece of text with a clickable substring.
+ * @param hyperlinkText The text within [linkText] that is clickable.
+ * @param linkURL The optional URL to return when the link is clicked.
+ * @param buttonText The text to show in the optional button.
+ * @param onLinkClick Invoked when the link is clicked. When not-null, the link will be visible.
+ * @param onButtonClick Invoked when the button is clicked. When not-null, the button will be visible.
+ * @param icon The UI to display before the text, typically an image or loading spinner.
+ */
+@Composable
+fun ReviewQualityCheckMessageCard(
+    title: String,
+    messageType: ReviewQualityCheckMessageType,
+    modifier: Modifier = Modifier,
+    description: String? = null,
+    linkText: String = "",
+    hyperlinkText: String = "",
+    linkURL: String = "",
+    buttonText: String = "",
+    onLinkClick: (() -> Unit)? = null,
+    onButtonClick: (() -> Unit)? = null,
+    icon: @Composable () -> Unit,
+) {
+    Card(
+        modifier = modifier,
+        shape = cardShape,
+        backgroundColor = messageType.cardBackgroundColor,
+        elevation = 0.dp,
+    ) {
+        Column(modifier = Modifier.padding(horizontal = 12.dp, vertical = 8.dp)) {
+            Row {
+                icon()
+
+                Spacer(modifier = Modifier.width(12.dp))
+
+                Column {
+                    Text(
+                        text = title,
+                        color = FirefoxTheme.colors.textPrimary,
+                        style = FirefoxTheme.typography.headline8,
+                    )
+
+                    description?.let {
+                        Spacer(modifier = Modifier.height(4.dp))
+
+                        Text(
+                            text = description,
+                            color = FirefoxTheme.colors.textPrimary,
+                            style = FirefoxTheme.typography.body2,
+                        )
+                    }
+
+                    onLinkClick?.let {
+                        Spacer(modifier = Modifier.height(4.dp))
+
+                        LinkText(
+                            text = linkText,
+                            linkTextState = LinkTextState(
+                                text = hyperlinkText,
+                                url = linkURL,
+                                onClick = { onLinkClick() },
+                            ),
+                            style = FirefoxTheme.typography.body2.copy(
+                                color = FirefoxTheme.colors.textPrimary,
+                            ),
+                            linkTextColor = FirefoxTheme.colors.textPrimary,
+                            linkTextDecoration = TextDecoration.Underline,
+                        )
+                    }
```

Comment:
Not sure if we can completely generalize it like this. There are some instances where the link is in its own line, but there's also instances where it's a "Learn more" in the description. As seen in [figma](https://www.figma.com/file/VE5xayz3yOSB5qOATUU9Im/Mobile-Shopping-Experience-MVP?node-id=5766%3A23503&mode=dev)

Reply:
In the latter scenario you describe, we'd just use link text instead. This `LinkText` Composable works where it gets two parameters: 1) The entire piece of text 2) the piece of text that is the link that needs the visual treatment. So we cover the use cases in the figma.